### PR TITLE
Issue #862 is fixed

### PR DESCRIPTION
**Pull request by onurd86**
Created this branch to fix issue #862 in which Evomaster gives error message about failure to create a directory after tests, not before.

### CODE REVIEW

File: core/src/test/kotlin/org/evomaster/core/config/ConfigPathCheckTest.kt [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+            EMConfig.validateOptions(args)
+        }
+
+        Assertions.assertEquals(exception.message, "Parameter 'outputFolder' refers to a file that does not exist, " +
+                "but the provided file path cannot be used to create a directory: $absoluteFilePath\n" +
+                "Please check file permissions of parent directories")
+
+
+        // delete path to write tests
+        filePathRoot.deleteRecursively()
```
Comment (arcuri82) [inline core/src/test/kotlin/org/evomaster/core/config/ConfigPathCheckTest.kt:60]: this should be in a `finally` block, where previous code after create directory should be in a `try`. if any exception is throw, should make sure that file is deleted once test is completed

File: core/src/test/kotlin/org/evomaster/core/config/ConfigPathCheckTest.kt [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+            IllegalArgumentException::class.java
+        ) {
+            EMConfig.validateOptions(args)
+        }
+
+        Assertions.assertEquals(exception.message, "Parameter 'outputFolder' refers to a folder that already " +
+                "exists, but that cannot be written to: $absolutePath")
+
+        // delete path to write tests
+        filePathRoot.deleteRecursively()
```
Comment (arcuri82) [inline core/src/test/kotlin/org/evomaster/core/config/ConfigPathCheckTest.kt:96]: same comment as before

File: core/src/test/kotlin/org/evomaster/core/config/ConfigPathCheckTest.kt [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+            IllegalArgumentException::class.java
+        ) {
+            EMConfig.validateOptions(args)
+        }
+
+        Assertions.assertEquals(exception.message, "Parameter 'outputFolder' refers to a file that already" +
+                " exists, but that it is not a folder: $absolutePath")
+
+        // delete path to write tests
+        filePathRoot.deleteRecursively()
```
Comment (arcuri82) [inline core/src/test/kotlin/org/evomaster/core/config/ConfigPathCheckTest.kt:134]: same comment as before

**Review by arcuri82 — CHANGES_REQUESTED** [review]

File: core/src/test/kotlin/org/evomaster/core/config/ConfigPathCheckTest.kt (hunk 1 above)
Comment (onurd86) [reply core/src/test/kotlin/org/evomaster/core/config/ConfigPathCheckTest.kt]: Hi Andrea,

Thanks for your comment. I have applied this comment to all test cases.

File: core/src/test/kotlin/org/evomaster/core/config/ConfigPathCheckTest.kt (hunk 2 above)
Comment (onurd86) [reply core/src/test/kotlin/org/evomaster/core/config/ConfigPathCheckTest.kt]: Hi Andrea,

Thanks for your comment. I have applied this comment for all test cases.

File: core/src/test/kotlin/org/evomaster/core/config/ConfigPathCheckTest.kt (hunk 3 above)
Comment (onurd86) [reply core/src/test/kotlin/org/evomaster/core/config/ConfigPathCheckTest.kt]: Hi Andrea,

Thanks for your comment. I have applied this comment to all test cases.

**Review by arcuri82 — APPROVED** [review]
#862

**Pull request merged**